### PR TITLE
Replace jQueryUI datetimepicker with jQuery Bootstrap datetimepicker

### PR DESCRIPTION
**Pull request by dalpo**
- Refactor `Datetime`, `Date` and `Time` type inputs
- Replace jquery ui datepicker with https://eonasdan.github.io/bootstrap-datetimepicker/
- Refactor `Datetime`, `Date` and `Time` search filters

### CODE REVIEW

**Comment by dalpo** [comment]
WIP: I'm fixing failing specs

**Comment by dalpo** [comment]
@mshibuya I'm not able to reproduce that failing spec... Any idea?

**Comment by dalpo** [comment]
The failing specs about mongoid are related to https://github.com/DatabaseCleaner/database_cleaner/pull/385

File: config/locales/rails_admin.en.yml [hunk 1]
```diff
@@ -27,7 +27,6 @@ en:
       next: "Next &raquo;"
       truncate: "…"
     misc:
-      filter_date_format: "mm/dd/yy" # a combination of 'dd', 'mm' and 'yy' with any delimiter. No other interpolation will be done!
```
Comment (mshibuya) [inline config/locales/rails_admin.en.yml]: This is incompatible change, please state that in changelog.

Comment (dalpo) [inline config/locales/rails_admin.en.yml]: Changelog updated! :smiley: 

Let me know if you have more feedback.
Thanks!

**Comment by dalpo** [comment]
Some screenshots:

screenshot from 2015-09-05 12 23 03

screenshot from 2015-09-05 12 25 01

**Comment by dalpo** [comment]
Everythink is Green! :fireworks: 

 @mshibuya, @sferik (or anyone else) could you review this PR an let me know what do you think about it?

File: lib/rails_admin/config/fields/types/bson_object_id.rb [hunk 2]
```diff
@@ -1,4 +1,9 @@
 require 'rails_admin/config/fields/types/string'
+begin
+  require 'mongoid'
+rescue LoadError => e
+  puts "[RailsAdmin] #{e.message}"
+end
```
Comment (mshibuya) [inline lib/rails_admin/config/fields/types/bson_object_id.rb]: This message is useless for non-Mongoid users.
Can't we just go without requiring mongoid here?

Comment (dalpo) [inline lib/rails_admin/config/fields/types/bson_object_id.rb]: You are right, we can remove it.

**Comment by mshibuya** [comment]
Great thanks for your effort! :smiley:

**Comment by fmh** [comment]
Good job 👍, but have this error :
````
No such file to load -- rails_admin/i18n_support
````

**Comment by dalpo** [comment]
Thanks guys :smile: 

@fmh Could you paste the stack trace?

**Comment by fmh** [comment]
"rails_admin/i18n_support" is required in application_helper
``````
/Users/fmh/.rvm/rubies/ruby-2.2.1/bin/ruby -e $stdout.sync=true;$stderr.sync=true;load($0=ARGV.shift) /Users/fmh/MyHomeCloud/Pro/Dev/E_app/bin/rails server thin -b 0.0.0.0 -p 3000 -e development
=> Booting Thin
=> Rails 4.2.3 application starting in development on http://0.0.0.0:3000
=> Run `rails server -h` for more startup options
=> Ctrl-C to shutdown server
Exiting
/Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require': No such file to load -- rails_admin/i18n_support (LoadError)
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `block in require'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:238:in `block in load_dependency'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:647:in `new_constants_in'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:238:in `load_dependency'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
	from /Users/fmh/MyHomeCloud/Pro/Dev/E_app/app/helpers/rails_admin/application_helper.rb:1:in `<top (required)>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:457:in `load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:457:in `block in load_file'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:647:in `new_constants_in'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:456:in `load_file'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:354:in `require_or_load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:317:in `depend_on'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:233:in `require_dependency'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/actionpack-4.2.3/lib/abstract_controller/helpers.rb:149:in `block in modules_for_helpers'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/actionpack-4.2.3/lib/abstract_controller/helpers.rb:144:in `map!'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/actionpack-4.2.3/lib/abstract_controller/helpers.rb:144:in `modules_for_helpers'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/actionpack-4.2.3/lib/action_controller/metal/helpers.rb:93:in `modules_for_helpers'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/actionpack-4.2.3/lib/abstract_controller/helpers.rb:108:in `helper'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/actionpack-4.2.3/lib/action_controller/railties/helpers.rb:17:in `inherited'
	from /Users/fmh/MyHomeCloud/Pro/Dev/E_app/app/controllers/application_controller.rb:2:in `<top (required)>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:457:in `load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:457:in `block in load_file'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:647:in `new_constants_in'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:456:in `load_file'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:354:in `require_or_load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:494:in `load_missing_constant'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:184:in `const_missing'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/inflector/methods.rb:261:in `const_get'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/inflector/methods.rb:261:in `block in constantize'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/inflector/methods.rb:259:in `each'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/inflector/methods.rb:259:in `inject'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/inflector/methods.rb:259:in `constantize'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/core_ext/string/inflections.rb:66:in `constantize'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/bundler/gems/rails_admin-50d6e59bc8bf/app/controllers/rails_admin/application_controller.rb:13:in `<module:RailsAdmin>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/bundler/gems/rails_admin-50d6e59bc8bf/app/controllers/rails_admin/application_controller.rb:3:in `<top (required)>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:457:in `load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:457:in `block in load_file'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:647:in `new_constants_in'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:456:in `load_file'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:354:in `require_or_load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:494:in `load_missing_constant'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:184:in `const_missing'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/bundler/gems/rails_admin-50d6e59bc8bf/app/controllers/rails_admin/main_controller.rb:2:in `<module:RailsAdmin>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/bundler/gems/rails_admin-50d6e59bc8bf/app/controllers/rails_admin/main_controller.rb:1:in `<top (required)>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:457:in `load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:457:in `block in load_file'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:647:in `new_constants_in'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:456:in `load_file'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:354:in `require_or_load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:494:in `load_missing_constant'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:184:in `const_missing'
	from /Users/fmh/MyHomeCloud/Pro/Dev/E_app/config/initializers/rails_admin_controllers.rb:2:in `<module:RailsAdmin>'
	from /Users/fmh/MyHomeCloud/Pro/Dev/E_app/config/initializers/rails_admin_controllers.rb:1:in `<top (required)>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:268:in `load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:268:in `block in load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:268:in `load'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/engine.rb:652:in `block in load_config_initializer'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/notifications.rb:166:in `instrument'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/engine.rb:651:in `load_config_initializer'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/engine.rb:616:in `block (2 levels) in <class:Engine>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/engine.rb:615:in `each'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/engine.rb:615:in `block in <class:Engine>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/initializable.rb:30:in `instance_exec'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/initializable.rb:30:in `run'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/initializable.rb:55:in `block in run_initializers'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:226:in `block in tsort_each'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:420:in `block (2 levels) in each_strongly_connected_component_from'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:429:in `each_strongly_connected_component_from'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:419:in `block in each_strongly_connected_component_from'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/initializable.rb:44:in `each'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/initializable.rb:44:in `tsort_each_child'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:413:in `call'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:413:in `each_strongly_connected_component_from'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:347:in `block in each_strongly_connected_component'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:345:in `each'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:345:in `call'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:345:in `each_strongly_connected_component'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:224:in `tsort_each'
	from /Users/fmh/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:203:in `tsort_each'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/initializable.rb:54:in `run_initializers'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/application.rb:352:in `initialize!'
	from /Users/fmh/MyHomeCloud/Pro/Dev/E_app/config/environment.rb:5:in `<top (required)>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `block in require'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.3/lib/active_support/dependencies.rb:274:in `require'
	from /Users/fmh/MyHomeCloud/Pro/Dev/E_app/config.ru:3:in `block in <main>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/rack-1.6.4/lib/rack/builder.rb:55:in `instance_eval'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/rack-1.6.4/lib/rack/builder.rb:55:in `initialize'
	from /Users/fmh/MyHomeCloud/Pro/Dev/E_app/config.ru:in `new'
	from /Users/fmh/MyHomeCloud/Pro/Dev/E_app/config.ru:in `<main>'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/rack-1.6.4/lib/rack/builder.rb:49:in `eval'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/rack-1.6.4/lib/rack/builder.rb:49:in `new_from_string'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/rack-1.6.4/lib/rack/builder.rb:40:in `parse_file'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/rack-1.6.4/lib/rack/server.rb:299:in `build_app_and_options_from_config'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/rack-1.6.4/lib/rack/server.rb:208:in `app'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/commands/server.rb:61:in `app'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/rack-1.6.4/lib/rack/server.rb:336:in `wrapped_app'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/commands/server.rb:139:in `log_to_stdout'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/commands/server.rb:78:in `start'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/commands/commands_tasks.rb:80:in `block in server'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/commands/commands_tasks.rb:75:in `tap'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/commands/commands_tasks.rb:75:in `server'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /Users/fmh/.rvm/gems/ruby-2.2.1/gems/railties-4.2.3/lib/rails/commands.rb:17:in `<top (required)>'
	from /Users/fmh/MyHomeCloud/Pro/Dev/E_app/bin/rails:4:in `require'
	from /Users/fmh/MyHomeCloud/Pro/Dev/E_app/bin/rails:4:in `<top (required)>'
	from -e:1:in `load'
	from -e:1:in `<main>'
```````

**Comment by dalpo** [comment]
Don't worry :wink: 

Let me know if you find some other kind of issue!

**Comment by fmh** [comment]
@dalpo thx for this feature , 
After some tests, I noticed some visual adjustments to make. look at the screenshots :
- date widget : very large
- time widget : I can not display it
- filter form : cannot filter with time field.
capture d ecran 2015-09-12 a 12 06 56
- in modal form : maybe some css settings to correct the display
capture d ecran 2015-09-12 a 12 07 59
I try to send a PR until tomorrow.

Sorry for my English 😊 🇬🇧

**Comment by dalpo** [comment]
Could you try with a vanilla rails_admin installation, without css customization?

**Comment by fmh** [comment]
this work fine 😳😳😳

**Comment by dalpo** [comment]
Thanks @fmh, you are right!

I'm going to check the following issues soon:

- [ ] Better html markup for time input
- [ ] Check time field filters


BTW I have a question... 
What kind of database datatype are you using to store `HH:mm` values?

**Comment by fmh** [comment]
DataType : Time 
Rails Timezone:  Bern
DB (mysql) Timezone : default

**Comment by dalpo** [comment]
@fmh I opened a new PR https://github.com/sferik/rails_admin/pull/2408

**Comment by SuperMasterBlasterLaser** [comment]
@dalpo Filtering by time only works when language is set to english. When I switched locale to russian, it does not filter by `date` and `between`. It just shows all available models in database.

I used Mongoid. Don't know if it is happening on AR.

**Pull request merged**